sycl/include/sycl/multi_ptr.hpp

```diff
@@ -312,9 +315,7 @@ class __SYCL_TYPE(multi_ptr) multi_ptr {
   explicit
   operator multi_ptr<value_type, OtherSpace, OtherIsDecorated>() const {
     return multi_ptr<value_type, OtherSpace, OtherIsDecorated>{
-        detail::cast_AS<typename multi_ptr<value_type, OtherSpace,
-                                           access::decorated::yes>::pointer>(
-            get_decorated())};
+        detail::static_address_cast<OtherSpace>(get_decorated())};
```

Comment:
Conversions from `generic_ptr` are defined such that it's UB if the pointer doesn't address the destination adddres space, same for cases below.